include/Framework/Process.h

```diff
@@ -133,6 +133,9 @@ class Process {
   }
 
  private:
+  /// The parameters used to configure this class.
+  framework::config::Parameters config_; 
```

Comment:
Boo persistence of the parameters object! I know it may be necessary, but I don't like it :frowning_face: 

Although... Maybe this copy of the parameters is persisted so that all other copies can be const-references. This would save some configuration time because we wouldn't be copying some many parameter objects.

Reply:
Yes, I don't like the fact that we are copying so many parameter objects.  However, this points to a deeper issue with how the framework as a whole is being configured which it outside the scope of this PR. 

I think the framework needs a major cleanup.  Process has become very difficult to read and the EventFile classes are kind of a mess.  Everything needs to be rewritten such that we modularize a lot of the behavior. Let's discuss ...